src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderGetMethodItemExtension.php

```diff
@@ -43,35 +54,44 @@ public function applyToItem(
         string $operationName = null,
         array $context = []
     ) {
-        $operationName = strtoupper($operationName);
-
         if (!is_a($resourceClass, OrderInterface::class, true)) {
             return;
         }
 
-        if ($operationName !== Request::METHOD_GET) {
+        if ($context[ContextKeys::HTTP_REQUEST_METHOD_TYPE] !== Request::METHOD_GET) {
             return;
         }
 
         $rootAlias = $queryBuilder->getRootAliases()[0];
         $user = $this->userContext->getUser();
 
-        $this->applyToItemForGetMethod($user, $queryBuilder, $operationName, $rootAlias);
+        $this->applyToItemForGetMethod($user, $queryBuilder, $rootAlias);
     }
 
     private function applyToItemForGetMethod(
         ?UserInterface $user,
         QueryBuilder $queryBuilder,
-        string $operationName,
         string $rootAlias
     ): void {
-        if ($user === null) {
+        /** @var string|null $cartCustomerId */
+        $cartCustomerId = $this->cartVisitorsCustomerContext->getCartCustomerId();
+
+        if ($user === null && $cartCustomerId === null) {
             $queryBuilder->andWhere(sprintf('%s.customer IS NULL', $rootAlias));
 
             return;
         }
 
-        if ($user instanceof ShopUserInterface && in_array('ROLE_API_ACCESS', $user->getRoles(), true)) {
+        if ($user === null && $cartCustomerId !== null) {
```

Comment:
Could you explain what you wanted to achieve? IMO you should add in query a statement to get only carts that do not have a customer or customer has no account

Reply:
In addressing step, when we are a visitor, customer is created automaticly : https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/ApiBundle/CommandHandler/Checkout/AddressOrderHandler.php#L75

Reply:
I understand, but that doesn't change anything from my comment above 😃 In this query extension, you should filter out orders/carts that visitor has no possibility to access (when there is a customer with user). The visitor is able to get to other orders, as long as they have a proper token value. So you don't need a customerId here, just write an appropriate query.